lib/vanagon/component.rb

```diff
@@ -204,7 +204,7 @@ def get_source(workdir) # rubocop:disable Metrics/AbcSize
         extract_with << source.extract(platform.tar) if source.respond_to? :extract
 
         @cleanup_source = source.cleanup if source.respond_to?(:cleanup)
-        @dirname = source.dirname
+        @dirname ||= source.dirname
```

Comment:
Well, this is to set `@dirname` if it isn't set by the user. So if it's already populated, then leave it as is. Otherwise, use the dirname provided by the source. I'm not sure the best way to do that other than the `||=`?